### DISCUSSION TITLE
support documents.get with tenant tokens

### DISCUSSION THREAD

**Discussion by derekperkins** [meilisearch/product#689] [Feedback & Feature Proposal]
We have several cases where we store ids and want to return values from meilisearch. The search to find those ids works great with tenant tokens, but we're unable to fetch those ids with tenant tokens, which is frustrating.
https://www.meilisearch.com/docs/reference/api/documents#get-documents-with-post

**Comment by macraig** [comment]
Hi @derekperkins , thanks for taking the time to give us your feedback, it's super valuable to us!

The team is quite busy working on other priorities, so I can't promise we can work on it at the moment. Can you accomplish this using the search endpoint as a workaround?

**Reply by derekperkins** [reply]
Yes, we were able to use search filters as a workaround.

**Comment by jameswagoner** [comment]
Our issue is with the hard search limits. Search endpoint worked fine until documents in an index (filtered by an tenant id) exceeded the limits.

We only leverage filters for this case and was happy to see the fetch documents endpoint accepts filters however learned that its not supported with tenant tokens (documents do not mention this BTW).

Tenants could have any number of documents in an index and meilisearch documents caution against setting high search limits as it would slow down performance.

**Reply by macraig** [reply]
Hi @jameswagoner, the search hard limits don't apply to the documents route, so they can't be the cause of any performance slowdowns. Would you be able to share your request including the filters used, the limit, and any other parameters you are using? This will help us investigate what might be causing your issues.

> documents do not mention this BTW

Thank you for flagging this, we'll work on making it more visible.

**Reply by jameswagoner** [reply]
Right, I just incorrectly assumed I would just all documents in the index like a DB query.

The comment about performance impact was from the docs about making changes to that search limit setting. Search had a nicer `page` and `perPage` feature and the tenant token applied the filter to limit results client side.

Since then, we are wrapping the fetch documents endpoint with a paginator and the tenant filter server side. Just a bummer to make an extra server hop.
